src/ar_http_iface_cowboy_handler.erl

```diff
@@ -0,0 +1,1142 @@
+%% TODO: Change so current block does not need to be pulled to calculate cost
+do_handle(<<"GET">>, [<<"price">>, SizeInBytes], _Req) ->
+	{200, #{},
+		integer_to_binary(
+			ar_tx:calculate_min_tx_cost(
+				binary_to_integer(SizeInBytes),
+				ar_node:get_current_diff(whereis(http_entrypoint_node))
+			)
+		)
+	};
+
+%% @doc Return the estimated reward cost of transactions with a data body size of 'bytes'.
+%% GET request to endpoint /price/{bytes}/{address}
+%% TODO: Change so current block does not need to be pulled to calculate cost
+do_handle(<<"GET">>, [<<"price">>, SizeInBytes, Addr], _Req) ->
+	case safe_decode(Addr) of
+		{error, invalid} ->
+			{400, #{}, <<"Invalid address.">>};
+		{ok, AddrOK} ->
+			{200, #{},
+				integer_to_binary(
+					ar_tx:calculate_min_tx_cost(
+						binary_to_integer(SizeInBytes),
+						ar_node:get_current_diff(whereis(http_entrypoint_node)),
+						ar_node:get_wallet_list(whereis(http_entrypoint_node)),
+						AddrOK
+					)
+				)
+			}
+	end;
+
+%% @doc Return the current hash list held by the node.
+%% GET request to endpoint /hash_list
+do_handle(<<"GET">>, [<<"hash_list">>], _Req) ->
+	HashList = ar_node:get_hash_list(whereis(http_entrypoint_node)),
+	{200, #{},
+		ar_serialize:jsonify(
+			ar_serialize:hash_list_to_json_struct(HashList)
+		)
+	};
+
+%% @doc Return the current wallet list held by the node.
+%% GET request to endpoint /wallet_list
+do_handle(<<"GET">>, [<<"wallet_list">>], _Req) ->
+	Node = whereis(http_entrypoint_node),
+	WalletList = ar_node:get_wallet_list(Node),
+	{200, #{},
+		ar_serialize:jsonify(
+			ar_serialize:wallet_list_to_json_struct(WalletList)
+		)
+	};
+
+%% @doc Share your nodes IP with another peer.
+%% POST request to endpoint /peers with the body of the request being your
+%% nodes network information JSON encoded as specified in ar_serialize.
+% NOTE: Consider returning remaining timeout on a failed request
+do_handle(<<"POST">>, [<<"peers">>], Req) ->
+	{BlockJSON, _} = read_complete_body(Req),
+	case ar_serialize:dejsonify(BlockJSON) of
+		{Struct} ->
+			{<<"network">>, NetworkName} = lists:keyfind(<<"network">>, 1, Struct),
+			case (NetworkName == <<?NETWORK_NAME>>) of
+				false ->
+					{400, #{}, <<"Wrong network.">>};
+				true ->
+					Peer = arweave_peer(Req),
+					case ar_meta_db:get({peer, Peer}) of
+						not_found ->
+							ar_bridge:add_remote_peer(whereis(http_bridge_node), Peer);
+						X -> X
+					end,
+					{200, #{}, []}
+			end;
+		_ -> {400, #{}, "Wrong network"}
+	end;
+%% @doc Return the balance of the wallet specified via wallet_address.
+%% GET request to endpoint /wallet/{wallet_address}/balance
+do_handle(<<"GET">>, [<<"wallet">>, Addr, <<"balance">>], _Req) ->
+	case safe_decode(Addr) of
+		{error, invalid} ->
+			{400, #{}, <<"Invalid address.">>};
+		{ok, AddrOK} ->
+			%% ar_node:get_balance/2 can time out which is not suitable for this
+			%% use-case. It would be better if it never timed out so that Elli
```

Comment:
One more Elli mention to remove.

Reply:
Is the comment up to date? Can I simply update the comment to read:
> It would be better if it never timed out so that Cowboy would handle the timeout instead.

Reply:
Yep